tools/crm_mon_print.c

```diff
@@ -208,7 +208,7 @@ print_resources(pcmk__output_t *out, pe_working_set_t *data_set,
             continue;
         }
 
-        if (!pe__rsc_running_on_any_node_in_list(rsc, only_show)) {
+        if (pcmk__rsc_is_filtered(rsc, only_show)) {
```

Comment:
I think there's one case where this shouldn't apply, if we're showing inactive resources and there's no host filter, then we should show all inactive resources regardless of whether they're allowed or known anywhere. I'm not sure how that would fit in with the other callers.

In other words, the current definition of pe__rsc_allowed_on_any_node_in_list() makes sense based on its name, but crm_mon needs that one case to be different.

Reply:
I've got a patch that should address this, but strangely it doesn't seem to make any difference in the test case output.  It seems like either this is already being handled in the code, or I don't have a test case that is showing the problem.

Reply:
It would be a corner case, primarily at cluster start-up, at resource creation, or immediately after a refresh, before the resource has been probed or started. Deleting all lrm_rsc_op entries for the resource should simulate it.

Reply:
Oh and it would also need target-role=Stopped or be banned from every node to not show up in allowed nodes

Reply:
Okay, there we go.  I think I see the problem (and the fix) now.